crates/configuration/src/serialized/native_procedure.rs

```diff
@@ -0,0 +1,82 @@
+use std::collections::BTreeMap;
+
+use mongodb::{bson, options::SelectionCriteria};
+use schemars::JsonSchema;
+use serde::Deserialize;
+
+use crate::schema::{ObjectField, ObjectType, Type};
+
+/// An arbitrary database command using MongoDB's runCommand API.
+/// See https://www.mongodb.com/docs/manual/reference/method/db.runCommand/
+///
+/// Native Procedures appear as "procedures" in your data graph.
+#[derive(Clone, Debug, Deserialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct NativeProcedure {
+    /// You may define object types here to reference in `result_type`. Any types defined here will
+    /// be merged with the definitions in `schema.json`. This allows you to maintain hand-written
+    /// types for native procedures without having to edit a generated `schema.json` file.
+    #[serde(default, skip_serializing_if = "BTreeMap::is_empty")]
+    pub object_types: BTreeMap<String, ObjectType>,
```

Comment:
I'm wondering whether all user-defined object types should be in one place, rather than each native procedure/query having their own. That would make it easier to re-use types between procedures/queries. It would also provide a place where users could override type definitions inferred by the schema inference.

Reply:
Yeah I think you make a good point. I think the way I have it set up has a convenience advantage because you can define single-use types in the same place they're used. But it might be more confusing. @codedmart already ran into this trying to write a procedure that reuses a collection object type.

I think this is a good topic for discussion. I think it's outside the scope of this PR since I already established this convention in native procedures. But there's still time to change it in another ticket.